src/map.cpp

```diff
@@ -3005,71 +2987,89 @@ MapBlock* ServerMap::loadBlock(v3s16 blockpos)
-		return getBlockNoCreateNoEx(blockpos);
-	}
-	// Not found in database, try the files
-
-	// The directory layout we're going to load from.
-	//  1 - original sectors/xxxxzzzz/
-	//  2 - new sectors2/xxx/zzz/
-	//  If we load from anything but the latest structure, we will
-	//  immediately save to the new one, and remove the old.
-	int loadlayout = 1;
-	std::string sectordir1 = getSectorDir(p2d, 1);
-	std::string sectordir;
-	if(fs::PathExists(sectordir1))
-	{
-		sectordir = sectordir1;
-	}
-	else
-	{
-		loadlayout = 2;
-		sectordir = getSectorDir(p2d, 2);
-	}
-
-	/*
-		Make sure sector is loaded
-	*/
-
-	MapSector *sector = getSectorNoGenerateNoEx(p2d);
-	if(sector == NULL)
-	{
-		try{
-			sector = loadSectorMeta(sectordir, loadlayout != 2);
-		}
-		catch(InvalidFilenameException &e)
+	} else {
+		// Not found in database, try the files
+
+		// The directory layout we're going to load from.
+		//  1 - original sectors/xxxxzzzz/
+		//  2 - new sectors2/xxx/zzz/
+		//  If we load from anything but the latest structure, we will
+		//  immediately save to the new one, and remove the old.
+		int loadlayout = 1;
+		std::string sectordir1 = getSectorDir(p2d, 1);
+		std::string sectordir;
+		if(fs::PathExists(sectordir1))
 		{
-			return NULL;
+			sectordir = sectordir1;
 		}
-		catch(FileNotGoodException &e)
+		else
 		{
-			return NULL;
+			loadlayout = 2;
+			sectordir = getSectorDir(p2d, 2);
 		}
-		catch(std::exception &e)
+
+		/*
+		Make sure sector is loaded
+		 */
+
+		MapSector *sector = getSectorNoGenerateNoEx(p2d);
+		if(sector == NULL)
 		{
-			return NULL;
+			try{
+				sector = loadSectorMeta(sectordir, loadlayout != 2);
+			}
+			catch(InvalidFilenameException &e)
+			{
+				return NULL;
+			}
+			catch(FileNotGoodException &e)
+			{
+				return NULL;
+			}
+			catch(std::exception &e)
+			{
+				return NULL;
+			}
 		}
-	}
 
-	/*
+		/*
 		Make sure file exists
-	*/
+		 */
 
-	std::string blockfilename = getBlockFilename(blockpos);
-	if(fs::PathExists(sectordir + DIR_DELIM + blockfilename) == false)
-		return NULL;
+		std::string blockfilename = getBlockFilename(blockpos);
+		if(fs::PathExists(sectordir + DIR_DELIM + blockfilename) == false)
+			return NULL;
 
-	/*
+		/*
 		Load block and save it to the database
-	*/
-	loadBlock(sectordir, blockfilename, sector, true);
-	return getBlockNoCreateNoEx(blockpos);
+		 */
+		loadBlock(sectordir, blockfilename, sector, true);
+	}
+	MapBlock *block = getBlockNoCreateNoEx(blockpos);
+	if (created_new && (block != NULL)) {
+		std::map<v3s16, MapBlock*> modified_blocks;
+		// Fix lighting if necessary
+		voxalgo::update_block_border_lighting(this, block, modified_blocks);
+		if(!modified_blocks.empty()) {
+			//Modified lighting, send event
+			MapEditEvent event;
+			event.type = MEET_OTHER;
+			std::map<v3s16, MapBlock *>::iterator it;
+			for (it = modified_blocks.begin();
+					it != modified_blocks.end(); ++it)
+			event.modified_blocks.insert(it->first);
```

Comment:
Needs indent.
I assume this line is the single line looped in the 'for' statement.